### ISSUE TITLE
`rule-name`: Reduce dangerous false-positives with type information

### ISSUE THREAD

**Issue by Samuel-Therrien-Beslogic** [sindresorhus/eslint-plugin-unicorn#2474]
### Description

Code using `appendChild` on a Node should not be autofixed to use `append`. I think this can be improved by using type information and only raising an issue (or at the very least not autofixing) unless it is known to be an Element.

This relates to https://github.com/sindresorhus/eslint-plugin-unicorn/issues/347

### Fail

These are currently working fine and should still work

```ts
const foo = (element: Element, htmlElement: HTMLElement) => {
  element.appendChild(node)
  htmlElement.appendChild(node)
}
```

```ts
document.getElementsByName("name")[0]?.appendChild(document.createTextNode("foo"))
```

### Pass

```ts
const foo = (node: Node, childNode: ChildNode, untyped,  element: Element) => {
  node.appendChild(node) // Current false positive
  childNode.appendChild(node) // Current false positive
  childNode.appendChild(node) // Current false positive
  element.append(node)
}
```

```ts
document.body.childNodes[0]?.appendChild(document.createTextNode("foo")) // Current false positive
```

**Comment by fregante** [comment]
A simplified example:

```ts
document.body.firstChild!.appendChild(new Text('a'));
// ESLint Error: Prefer `Node#append()` over `Node#appendChild()`.
```

and then:

```ts
document.body.firstChild!.append(new Text('a'));
// TypeScript Error: Property 'append' does not exist on type 'ChildNode'.
```

So yeah it is true that `appendChild` works where `append` doesn't, but that actually is a footgun. Try this [perfectly valid](https://www.typescriptlang.org/play/?#code/HYUw7gBAKiAeAuAKAlAOgIYAdMmAEwGEALASwBs9FRIYEVkg) TypeScript code:

```ts
new Text().appendChild(new Text());
// HierarchyRequestError: The operation would yield an incorrect node tree.
```

That's a runtime error.

So yeah, `prefer-modern-dom-apis` is technically wrong for suggesting `append` where it doesn't exist, but if you follow the advice fully it will save you from runtime errors, which means the rule is still valid as is.

Perhaps what I'd suggest is expanding the error/docs to include this piece of information _and_ to fix the error (no such thing as `Node#append`, it's `Element#append`)

**Comment by sindresorhus** [comment]
> Perhaps what I'd suggest is expanding the error/docs to include this piece of information and to fix the error (no such thing as Node#append, it's Element#append)

👍
